src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -266,16 +264,17 @@ async def test_TC_MCORE_FS_1_4(self):
             filter=discriminator,
         )
 
-        # Wait some time, so the dynamic endpoint will appear on the TH_FSA_BRIDGE.
-        await asyncio.sleep(5)
-
-        # Get the list of endpoints on the TH_FSA_BRIDGE after adding the TH_SERVER_NO_UID.
-        th_fsa_bridge_endpoints_new = set(await self.read_single_attribute_check_success(
-            cluster=Clusters.Descriptor,
-            attribute=Clusters.Descriptor.Attributes.PartsList,
-            node_id=th_fsa_bridge_th_node_id,
-            endpoint=0,
-        ))
+        th_fsa_bridge_endpoints_new = set(th_fsa_bridge_endpoints)
+        # Try to get the dynamic endpoint number for the TH_SERVER_NO_UID on the TH_FSA_BRIDGE.
+        while th_fsa_bridge_endpoints_new == th_fsa_bridge_endpoints:
+            await asyncio.sleep(0.5)
+            # Get the list of endpoints on the TH_FSA_BRIDGE.
+            th_fsa_bridge_endpoints_new.update(await self.read_single_attribute_check_success(
+                cluster=Clusters.Descriptor,
+                attribute=Clusters.Descriptor.Attributes.PartsList,
+                node_id=th_fsa_bridge_th_node_id,
+                endpoint=0,
+            ))
```

Comment:
nit, do we want to timeout as some point? Previously I think using the `expected_output=` has some default timeout. This does not look like it has a timeout

Reply:
I was thinking about that, but in general the entire test has a timeout... if you want I can add something like ~30 seconds (i.e. 100 iterations)

Reply:
Added 60 iterations timeout.